webgpu.h

```diff
@@ -1675,6 +1686,8 @@ WGPU_EXPORT void wgpuDeviceRelease(WGPUDevice device) WGPU_FUNCTION_ATTRIBUTE;
 
 // Methods of Instance
 WGPU_EXPORT WGPUSurface wgpuInstanceCreateSurface(WGPUInstance instance, WGPUSurfaceDescriptor const * descriptor) WGPU_FUNCTION_ATTRIBUTE;
+WGPU_EXPORT size_t wgpuInstanceEnumerateWGSLLanguageFeatures(WGPUInstance instance, WGPUWGSLFeatureName * features) WGPU_FUNCTION_ATTRIBUTE;
```

Comment:
We may want to make the change we decided on in #252 before landing this. Otherwise we'll add this and then change it soon.

That said, having considered what we'd change here, I'm going to comment there about whether that's actually what we want to do.

Reply:
When do you think these changes will be made in webgpu.h?

We're going to ship ReadonlyAndReadwriteStorageTextures, Packed4x8IntegerDotProduct, UnrestrictedPointerParameters, and PointerCompositeAccess in Chrome 122 and later and it'd be nice if developers can check whether those WGSL language features are supported or not.

Reply:
How about I remove `wgpuInstanceEnumerateWGSLLanguageFeatures` from this CL and only add `wgpuInstanceHasWGSLLanguageFeature `? What do you think?

Reply:
Most often webgpu.h changes after Dawn changes - Dawn implements them in our generator (even if only behind the "upstream" tag) and then we "roll" it into this repository.

Adding just HasWGSLLanguageFeature for now would be fine, then we can land most of this PR and deal with the rest soon after.

I'll also note it is not critical for changes to land here before they are implemented in Emscripten. Emscripten actually tracks the Dawn API and not the upstream webgpu.h API right now, for practical reasons for people cross-targeting Dawn and Emscripten.

Still to avoid downstream projects depending on the wgpuInstanceEnumerateWGSLLanguageFeatures API before we fix it, I think Emscripten would also best implement only HasWGSLLanguageFeatures.

Reply:
Thanks for the detailed answer @kainino0x 
I've updated this PR and sent a PR for Emscripten at https://github.com/emscripten-core/emscripten/pull/21055